nntrainer/tensor/tensor.h

```diff
@@ -188,7 +194,51 @@ class Tensor {
    * @param[in] w width location
    */
   float getValue(unsigned int batch, unsigned int c, unsigned int h,
-                 unsigned int w) const;
+                 unsigned int w) const noexcept {
+    return getData()[getIndex(batch, c, h, w)];
+  }
+
+  /**
+   * @brief Get the Value thinking that it is padded
+   * for example, for the tensor (virtually padded) below,
+   * getValue(0, 0, 2, 2, 1, 1, .0f) will return 5
+   * padding available for height and width axis for now
+   * 0 0 0 0 0
+   * 0 1 2 3 0
+   * 0 4 5 6 0
+   * 0 7 8 9 0
+   * 0 0 0 0 0
+   * @param b batch index
+   * @param c channel index
+   * @param h height index
+   * @param w width index
+   * @param ph padding height
+   * @param pw padding width
+   * @return float value
+   */
+  float getValuePadded(unsigned int b, unsigned int c, unsigned int h,
+                       unsigned int w, unsigned int ph, unsigned int pw,
+                       float pad_value = 0) const EXCEPT_WHEN_DEBUG {
+#if DEBUG
+    unsigned int padded_h = 2 * ph + h;
+    unsigned int padded_w = 2 * pw + w;
+    if (h > padded_h && w > padded_w) {
+      throw std::out_of_range(
+        "[Tensor::getValuePadded] trying to access out of range");
+    }
+#endif
+
+    auto in_range = [](unsigned int virtual_pos, unsigned int pad,
+                       unsigned int actual_len) -> bool {
+      return pad <= virtual_pos && virtual_pos < (pad + actual_len);
+    };
+
+    if (in_range(h, ph, height()) && in_range(w, pw, width())) {
```

Comment:
how about move in_range check to DEBUG mode?

Reply:
in_range check is necessary to check because else it is returning pad_value

Reply:
aha, this creates virtual pad. I thought, this function treats the internal data as though it was actually padded and the user asked for some data.
You can consider renaming to `getValuePaddedVritual()`

Reply:
Yes. renamed to `getValuePaddedVirtual()` :)